pymeasure/instruments/instrument.py

```diff
@@ -149,134 +257,200 @@ def binary_values(self, command, header_bytes=0, dtype=np.float32):
-                          and returns a valid value, while it otherwise raises an exception
+            and returns a valid value, while it otherwise raises an exception
         :param values: A list, tuple, range, or dictionary of valid values, that can be used
-                       as to map values if :code:`map_values` is True.
+            as to map values if :code:`map_values` is True.
         :param map_values: A boolean flag that determines if the values should be
-                          interpreted as a map
+            interpreted as a map
         :param get_process: A function that take a value and allows processing
-                            before value mapping, returning the processed value
+            before value mapping, returning the processed value
         :param set_process: A function that takes a value and allows processing
-                            before value mapping, returning the processed value
+            before value mapping, returning the processed value
+        :param command_process: A function that take a command and allows processing
+            before executing the command, for getting
         :param check_set_errors: Toggles checking errors after setting
         :param check_get_errors: Toggles checking errors after getting
+        :param dynamic: Specify whether the property parameters are meant to be changed in
+            instances or subclasses.
+
+        Example of usage of dynamic parameter is as follow:
+
+        .. code-block:: python
+
+            class GenericInstrument(Instrument):
+                center_frequency = Instrument.control(
+                    ":SENS:FREQ:CENT?;", ":SENS:FREQ:CENT %e Hz;",
+                    " A floating point property that represents the frequency ... ",
+                    validator=strict_range,
+                    # Redefine this in subclasses to reflect actual instrument value:
+                    values=(1, 26.5e9),
+                    dynamic=True  # declare property dynamic
+                )
+
+            class SpecificInstrument(Instrument):
+                # Identical to GenericInstrument, except for frequency range
+                center_frequency_values = (1, 13e9) # Redefined at subclass level
+
+            instrument = SpecificInstrument()
+            instrument.center_frequency_values = (1, 6e9) # Redefined at instance level
+
+        .. warning:: Unexepected side effects when using dynamic properties
+
+        Users must pay attention when using dynamic properties, since definition of class and/or
+        instance attributes matching specific patterns could have unwanted side effect.
+        The attribute name pattern `property_param`, where `property` is the name of the dynamic
+        property (e.g. `center_frequency` in the example) and `param` is any of this method
+        parameters name except `dynamic` (e.g. `values` in the example) has to be considered
+        reserved for dynamic property control.
```

Comment:
Sorry I missed this due to github "features": worth having a look in more detail. Thanks

Reply:
Actually I possibly thought about this, (sorry it was more than a year go, see https://github.com/pymeasure/pymeasure/pull/376#issuecomment-748640030).
What you proposed, I believe is possible at instance level only, but not during class definition, so examples in https://github.com/pymeasure/pymeasure/pull/376#issuecomment-1010294612 cannot be addressed.

In case, you still think this is possible also at class level, could You please send a code snippet to show the principle ?

Reply:
I'll have a think on it and see if I can prototype something.

Reply:
I have tried a couple of approaches, but it's as you say, I could not make this work with class definitions. That it, I could not find a nice way to mutate a class attribute `Instrument.dynamic_overrides` in a subclass to add e.g. values entries. The blocking problem is the existence (or lack thereof) of class attributes at subclass definition time.
I don't like that we strew attributes with weird and nearly-duplicated names through our Instruments, but I don't see a better way. I think we can now proceed to the documentation & updating of affected instrument implementations.

Reply:
For reference, https://stackoverflow.com/questions/9638921/can-instances-get-their-own-copy-of-class-level-mutable-defaults-without-using looked pretty promising regarding updates not affecting superclass entries, and a [data/non-data descriptor](https://docs.python.org/3/howto/descriptor.html#descriptor-protocol) seems like a probably approach for `Instrument.dynamic_overrides` (we're using something similar already in . So, if we find a better way in the future, or a smaller-scoped way (e.g. hide all the "special" variables we can revisit this.